sdk/resourcemanager/Azure.ResourceManager.Core/src/GenericResourceOperations.cs

```diff
@@ -61,8 +50,7 @@ public ArmResponse<Response> Delete(CancellationToken cancellationToken = defaul
         /// <returns> A <see cref="Task"/> that on completion returns the status of the delete operation. </returns>
         public async Task<ArmResponse<Response>> DeleteAsync(CancellationToken cancellationToken = default)
         {
-            _apiVersion = await ClientOptions.ApiVersions.TyrGetApiVersionAsync(ProviderOperations, Id, cancellationToken).ConfigureAwait(false);
-            var operation = await Operations.StartDeleteByIdAsync(Id, _apiVersion, cancellationToken).ConfigureAwait(false);
+            var operation = await Operations.StartDeleteByIdAsync(Id,  await ClientOptions.ApiVersions.TryGetApiVersionAsync(Id.Type, cancellationToken).ConfigureAwait(false), cancellationToken).ConfigureAwait(false);
```

Comment:
I think we still need to store the result of TryGetApiVersionAsync in a stack variable and if it returns null throw InvalidOperationException saying unable to determine api version for X please set it in your client options

Reply:
We should NEVER get a null back in this use case. We will either return something or an exception will get thrown when the providers API is called for an invalid resource.

Reply:
true as long as GenericResourceOperations maintains only internal constructors, which isn't gauranteed at this point with the fluctuating API shape.

If you use the internal constructor and pass in "/subscriptions/{validGuid/providers/Microsoft.Compute/virtualToasters/mytoaster" that will load just fine and then TryGetApiVersion will return null

in fact lets add this test case to assert the throw